src/librustc/hir/mod.rs

```diff
@@ -1929,22 +1929,30 @@ pub struct PolyTraitRef {
     pub span: Span,
 }
 
+pub type Visibility = Spanned<Visibility_>;
+
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
-pub enum Visibility {
-    Public,
-    Crate(CrateSugar),
-    Restricted { path: P<Path>, id: NodeId },
-    Inherited,
+pub enum Visibility_ {
+    VisibilityPublic,
+    VisibilityCrate(CrateSugar),
+    VisibilityRestricted { path: P<Path>, id: NodeId },
+    VisibilityInherited,
```

Comment:
The convention `enum Something_ { SomethingVariant }` is "deprecated" and exists in HIR only because not all HIR structures were updated to the "modern" form `enum SomethingKind { Variant }` like it was done in AST.

New structures should not use it, so it'd be better to rename `Visibility_` to `VisibilityKind`.